prepare/kotlin-intellij-test/build.gradle.kts

```diff
@@ -0,0 +1,49 @@
+plugins {
```

Comment:
Have you tried to use artifacts provided by intellij instead of packing them here? I think it might be more convenient
https://plugins.jetbrains.com/docs/intellij/intellij-artifacts.html

Reply:
I tried that but I was unable to get it to work

Reply:
So the idea is that we write scripts to pull those artifacts down into a local directory and then integrate it with gradle by registering a local repository? It seems like this is more convenient to have in the compiler since we can then guarantee that the versions are compatible and we'd be replicating all the dependency pulling code. If this is an unacceptable solution though I will start writing the scripts.

Reply:
I meant you could put those artifact to the test runtime directly from idea repo without any additional processing.
I was assuming this artifact is runtime dependency of the test framework, may be I misunderstood it, could you show use site script sample?

Reply:
`testImplementation("org.jetbrains.kotlin:kotlin-intellij-test:$ktVersion")`

Reply:
Then it should be possible to add artifacts directly from intellij repo instead of building and publishing a fat jar. For instance instead of doing `embedded(intellijDep()) { includeJars("jps-model") }` it should be possible to do `testImplementation("com.jetbrains.intellij.platform:jps-model-impl:some-version")` on use site. Or even better add all those Idea libraries to the POM of the test framework artifact so Gradle will resolve and download them without any additional artifacts.

Reply:
`testImplementation("com.jetbrains.intellij.platform:jps-model:some-version")` will not work because that jar lives in the com.jetbrains.intellij.idea.ideaIC zip. You have to add '"com.jetbrains.intellij.idea:ideaIC:$intellijVersion"' to a configuration then unzip it before grabbing specific jars like jps model. I can probably figure out how to do this by looking at gradle examples

Reply:
All those jars also live in the separate maven repository and could be used without any unpacking right away from that repo. It is described here https://plugins.jetbrains.com/docs/intellij/intellij-artifacts.html

Reply:
Ok by including the following:
```
val platform_dependencies = listOf(
        "util",
        "jps-model",
        "test-framework-core",
        "util-rt",
        "util-strings",
        "util-text-matching",
        "util-diagnostic",
        "util-collections",
        "util-class-loader",
        "statistics-config",
        "jps-model-impl",
        "jps-model-serialization",
        "jps-build",
        "jps-build-javac-rt")
dependencies {
    platform_dependencies.forEach {
        testImplementation("com.jetbrains.intellij.platform:$it:$intelliJVersion")
    }
}

```
I was able to retrieve the intellij dependencies but now it's broken on `Can't find bundle for base name messages.JavaPsiBundle, locale en_US`. I'll audit my dependencies

Reply:
That worked for the platform dependencies but there are some dependencies in intellij-core that are not available in the cache. I tried adding all available modules from that site and still had some missing dependency errors. I tracked down the source of the missing classes as intellij-core. I'm going to try and add a task that will grab the archive and unzip it into the correct location

Reply:
Is there additional configuration required for using intellij dependencies? Program failed at runtime on a thrown exception `Caused by: java.util.MissingResourceException: Can't find bundle for base name messages.JavaPsiBundle, locale en_US`. Are you familiar with that error?

Reply:
yep, got it working thanks!